client/src/components/ToolInformation/services.js

```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
+import { getAppRoot } from "onload/loadConfig";
+import { rethrowSimple } from "utils/simple-error";
+
+export async function sendErrorReportTool(dataset, message, email, transcript) {
```

Comment:
We should be using the typescript fetcher stuff for this and not axios. Want me to take a pass at adjusting that for you? Do you have a setup for testing this?

Reply:
Sure, it would be great to see an example (or even a scaffold) of what you had in mind with the "fetcher" amongst the existing code. Let me first push another update of the code so you have something more functional (will post back here when ready).

Reply:
@jmchilton Code updated and ready for your suggestions / adjustments.

Reply:
I've made a modification to this file with https://github.com/galaxyproject/galaxy/pull/17969 - that updates the older existing axios call to use a typescript fetcher. I think you'll need to rebuild your schema for your new API with ``make update-client-api-schema`` from Galaxy's root and then create a typed fetcher for your new API path - it will be something like ``const errorReporter = fetcher.path("/api/user-reporting/error").method("post").create();``. If everything is setup, your IDE should be able to do typed checks of how you call ``errorReporter`` that it cannot do with ``axios`` to ensure the API is only supplied with valid parameters.